doc/optimizers.md

```diff
@@ -1,3 +1,59 @@
+## AdaBound
```

Comment:
I just remembered, we should add `AdaBound` and `AMSBound` to the list of optimizers that can be used with separable differentiable functions in `doc/function_types.md`. :+1: